cnxepub/tests/test_models.py

```diff
@@ -134,6 +134,22 @@ def test_sanitize_xml(self):
         xml = target(content, recover=False)
         self.assertEqual(xml, '<img src="aaa_bbb.png"/>')
 
+    def test_sanitize_xml_not_html(self):
+        # The serializer used lxml.html.create_fragments which caused truncation
+        from ..models import _sanitize_xml as target
+        test_dir = os.path.dirname(__file__)
+
+        source_file = open(os.path.join(test_dir, './test_models_source.xml'), 'r')
+        source = source_file.read().decode('utf-8')
+        source_file.close()
```

Comment:
@philschatz, Future reference...

```python
with open('file.txt', 'r') as file:
    file.read()
```
the with block will auto close the file. But this works just fine.

Reply:
Thanks! I knew about `with` but wasn't sure if it was a python3 thing or not. A quick search of how to read a file in 2.7 resulted in the 3-steps included in the PR. I'll update the PR 😄